crates/starknet-rs-py/src/cached_state.rs

```diff
@@ -46,9 +57,17 @@ impl PyCachedState {
             .to_biguint())
     }
 
-    fn set_contract_class(&mut self, _address: BigUint, _contract_class: &PyContractClass) {
-        // self.state.set_contract_class(class_hash, contract_class)
-        todo!()
+    fn set_contract_class(
+        &mut self,
+        address: BigUint,
+        contract_class: &PyContractClass,
+    ) -> PyResult<()> {
+        let address = felt_to_hash(&Felt::from(address));
```

Comment:
`address` would be the class hash right? Also, we can use `ClassHash` instead of `BigUint`, as it's just an alias for an array.